### PR TITLE
fix(ghost_text): generate it using chars instead of bytes

### PR DESCRIPTION
**Pull request by muniter**
Previously if the completion entry text had a multibyte character the
ghostly text renderd improperly, for example:

The cursor position is "|"

### 1. Example
In this case the ghost text was spliting a multibyte character.

screenshot_20211103_164346
screenshot_20211103_164803

### 2. Example
In this case the multibyte had passed messing up the indexing therefore presentin the last character of the entry as ghost text.

screenshot_20211103_164332
screenshot_20211103_164815

### CODE REVIEW

File: lua/cmp/view/ghost_text_view.lua [hunk 1]
```diff
@@ -51,6 +47,20 @@ ghost_text_view.new = function()
...
+ghost_text_view.text_gen = function(self, line, cursor_col)
+  local text
+  local cword = string.sub(line, self.entry:get_offset(), cursor_col)
+  local cword_char_len = vim.fn.strchars(cword)
+  local entry_text = self.entry:get_insert_text()
+  if self.entry.completion_item.insertTextFormat == types.lsp.InsertTextFormat.Snippet then
+    entry_text = vim.lsp.util.parse_snippet(entry_text)
+  end
+  -- Missing characters to complete the entry text
+  text = vim.fn.strcharpart(str.oneline(entry_text), cword_char_len)
```
Comment (hrsh7th) [inline lua/cmp/view/ghost_text_view.lua:60]: Could you use `vim.str_byteidx` instead of `vim.fn.strcharpart`?

Comment (muniter) [reply lua/cmp/view/ghost_text_view.lua]: Done

**Comment by muniter** [comment]
@hrsh7th Changed the implementation to only use `vim.str_{utf,byte}index`.

Also heads up on the following just merged functions in neovim that will be very helpful for more features in cmp.

- https://github.com/neovim/neovim/pull/16129
- https://github.com/neovim/neovim/pull/16124

**Comment by hrsh7th** [comment]
LGTM. Thank you~!

**Pull request merged**